### ISSUE TITLE
Yarn 2 support without Zero-Install

### ISSUE THREAD

**Issue by brandonchinn178** [heroku/heroku-buildpack-nodejs#884]
**Describe the bug**
Just to be clear, is Yarn 2 supported without zero-install set up (i.e. checking in `.pnp.js`, `.yarn/cache/`, etc.)? The docs are unclear on this point, and Yarn 2 equally supports both with and without zero-install.

My deployment is currently failing in my release step which runs `yarn ts-node -s bin/migrate.ts`
```
/app/.pnp.js:31343
    throw firstError;
    ^
Error: Qualified path resolution failed - none of the candidates can be found on the disk.
Source path: /app/.yarn/$$virtual/ts-node-virtual-3d03334661/2/yarncache.PF4oI/ts-node-npm-9.0.0-7f58a021c2-49d2ab087f.zip/node_modules/ts-node/dist/bin.js
Rejected candidate: /app/.yarn/$$virtual/ts-node-virtual-3d03334661/2/yarncache.PF4oI/ts-node-npm-9.0.0-7f58a021c2-49d2ab087f.zip/node_modules/ts-node/dist/bin.js
Rejected candidate: /app/.yarn/$$virtual/ts-node-virtual-3d03334661/2/yarncache.PF4oI/ts-node-npm-9.0.0-7f58a021c2-49d2ab087f.zip/node_modules/ts-node/dist/bin.js.js
Rejected candidate: /app/.yarn/$$virtual/ts-node-virtual-3d03334661/2/yarncache.PF4oI/ts-node-npm-9.0.0-7f58a021c2-49d2ab087f.zip/node_modules/ts-node/dist/bin.js.json
Rejected candidate: /app/.yarn/$$virtual/ts-node-virtual-3d03334661/2/yarncache.PF4oI/ts-node-npm-9.0.0-7f58a021c2-49d2ab087f.zip/node_modules/ts-node/dist/bin.js.node
    at internalTools_makeError (/app/.pnp.js:31087:34)
    at resolveUnqualified (/app/.pnp.js:32121:13)
    at resolveRequest (/app/.pnp.js:32145:14)
    at Object.resolveRequest (/app/.pnp.js:32217:26)
    at Function.external_module_.Module._resolveFilename (/app/.pnp.js:31320:34)
    at Function.external_module_.Module._load (/app/.pnp.js:31185:48)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:60:12)
    at internal/main/run_main_module.js:17:47
```

**To Reproduce**

**Versions (please complete the following information):**
 - Heroku Stack: heroku-20
 - Node Version: see below
 - NPM or Yarn Version: see below
 - Buildpack Version: `heroku/nodejs`, not pinned
```
-----> Installing binaries
       engines.node (package.json):  unspecified
       engines.npm (package.json):   unspecified (use default)
       engines.yarn (package.json):  unspecified (use default)
       
       Resolving node version 12.x...
       Downloading and installing node 12.20.0...
       Using default npm version: 6.14.8
       Resolving yarn version 1.22.x...
       Downloading and installing yarn (1.22.10)
       Using yarn 2.3.3-git.20201116.5b1bdc67
```

**Additional context**

**Comment by brandonchinn178** [comment]
Yes, it seems when my `release` step is run, `.yarn/cache/` doesn't exist.

**Comment by danielleadams** [comment]
Hi @brandonchinn178 - Yarn on Heroku is built to support with and without zero-install set up. Is the release step you're talking about the Heroku release step that is post build? Or is this a release script that happens within the build?

**Comment by brandonchinn178** [comment]
I'm referring to the `release` step in my `Procfile` (https://devcenter.heroku.com/articles/release-phase). I just tried the following steps again:

1. Remove my custom buildpack
1. Add `heroku/nodejs` as a buildpack
1. Redeploy `master` branch (which worked previously)

Build log shows:
```
-----> Build
       
-----> Build succeeded!
-----> Discovering process types
       Procfile declares types -> release, web
-----> Compressing...
       Done: 42M
-----> Launching...
 !     Release command declared: this new release will not be available until the command succeeds.
       Released v16
       https://worship-mate-graphql.herokuapp.com/ deployed to Heroku
```
Release log shows
```
/app/.pnp.js:31343
    throw firstError;
    ^
Error: Qualified path resolution failed - none of the candidates can be found on the disk.
Source path: /app/.yarn/$$virtual/ts-node-virtual-3d03334661/2/yarncache.TSjig/ts-node-npm-9.0.0-7f58a021c2-49d2ab087f.zip/node_modules/ts-node/dist/bin.js
Rejected candidate: /app/.yarn/$$virtual/ts-node-virtual-3d03334661/2/yarncache.TSjig/ts-node-npm-9.0.0-7f58a021c2-49d2ab087f.zip/node_modules/ts-node/dist/bin.js
Rejected candidate: /app/.yarn/$$virtual/ts-node-virtual-3d03334661/2/yarncache.TSjig/ts-node-npm-9.0.0-7f58a021c2-49d2ab087f.zip/node_modules/ts-node/dist/bin.js.js
Rejected candidate: /app/.yarn/$$virtual/ts-node-virtual-3d03334661/2/yarncache.TSjig/ts-node-npm-9.0.0-7f58a021c2-49d2ab087f.zip/node_modules/ts-node/dist/bin.js.json
Rejected candidate: /app/.yarn/$$virtual/ts-node-virtual-3d03334661/2/yarncache.TSjig/ts-node-npm-9.0.0-7f58a021c2-49d2ab087f.zip/node_modules/ts-node/dist/bin.js.node
    at internalTools_makeError (/app/.pnp.js:31087:34)
    at resolveUnqualified (/app/.pnp.js:32121:13)
    at resolveRequest (/app/.pnp.js:32145:14)
    at Object.resolveRequest (/app/.pnp.js:32217:26)
    at Function.external_module_.Module._resolveFilename (/app/.pnp.js:31320:34)
    at Function.external_module_.Module._load (/app/.pnp.js:31185:48)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:60:12)
    at internal/main/run_main_module.js:17:47
```
If I change my `Procfile` to say something like
```
release: find / && <normal command>
web: <normal command>
```
I see in the release logs that `/app/.yarn/cache/` doesn't exist

**Comment by schneems** [comment]
Internal link: Related to support ticket https://heroku.support/1022537

**Comment by brandonchinn178** [comment]
@joshwlewis hm seems like after rebasing against the latest master my PR no longer fixes the problem. I've reverted my fork to fork off of the state of the repo on Jan 6.

Using `https://github.com/heroku/heroku-buildpack-nodejs#ca735eb8b92e03b2ac6545595073215307aa20b9` as my buildpack, I once again see that `.yarn/cache/` no longer exists, possibly due to https://github.com/heroku/heroku-buildpack-nodejs/pull/891 ([specifically this line](https://github.com/heroku/heroku-buildpack-nodejs/pull/891/files#diff-a8cd9433d3fbaa8ab408ac28c1cc5578404377dd8cf3ae1c6705e6290d34fbd3R180-R184)). (Possibly related: https://github.com/heroku/heroku-buildpack-nodejs/issues/918).

**Comment by lizthegrey** [comment]
@brandonchinn178 are you still maintaining that fork? it seems to be deleted, and I'm running into the same problem with enabling pnp.

**Comment by brandonchinn178** [comment]
@lizthegrey the PR was merged, which contained the changes in my fork. If you're using the newest version of heroku-buildpack-nodejs, you should have the fix. Your error message seems different than the original error in the issue, so it's probably worth a separate ticket

**Comment by lizthegrey** [comment]
Fixed by #978

**Comment by joshwlewis** [comment]
I believe this issue to be fixed now that #978 is released.
